activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -418,21 +418,15 @@ def change_column_null(table_name, column_name, null, default = nil) #:nodoc:
           exec_query("UPDATE #{quote_table_name(table_name)} SET #{quote_column_name(column_name)}=#{quote(default)} WHERE #{quote_column_name(column_name)} IS NULL")
         end
         alter_table(table_name) do |definition|
-          definition[column_name].null = null
+          definition[column_name].options[:null] = null
```

Comment:
These cases of modifying a data collection directly feel like a step back from calling specific writer methods.